res/skins/LateNight/samplers_rack.xml

```diff
@@ -1,43 +1,68 @@
 <Template>
+
   <WidgetGroup>
     <Layout>horizontal</Layout>
     <SizePolicy>me,max</SizePolicy>
+    <Connection>
+      <ConfigKey>[Samplers],show_samplers</ConfigKey>
+      <BindProperty>visible</BindProperty>
+    </Connection>
     <Children>
 
-      <Template src="skin:sampler.xml">
-        <SetVariable name="samplernum">1</SetVariable>
-      </Template>
-
-      <WidgetGroup><Size>10f,1min</Size></WidgetGroup>
-
-      <Template src="skin:sampler.xml">
-        <SetVariable name="samplernum">2</SetVariable>
-      </Template>
-
-      <WidgetGroup><Size>10f,1min</Size></WidgetGroup>
+      <WidgetGroup>
+      <Layout>horizontal</Layout>
+        <Children>
+          <!-- Select between 1-4 big or orthers -->
+          <WidgetStack currentpage="[Master],show_4sampler">
+            <Children>
 
-      <Template src="skin:sampler.xml">
-        <SetVariable name="samplernum">3</SetVariable>
-      </Template>
+             <!-- Select between 5-8 big or 8 tiny -->
+                  <WidgetStack currentpage="[Samplers],show_big">
+                    <Children>
+                      <WidgetGroup>
+                        <Layout>horizontal</Layout>
+                        <Children>
+                          <Template src="skin:samplers_unit_1-8_big.xml"/>
+                        </Children>
+                      </WidgetGroup>
 
-      <WidgetGroup><Size>10f,1min</Size></WidgetGroup>
+                      <!--8 tiny Sampler -->
+                      <WidgetGroup trigger="[Samplers],show_big5678_big_of_8" on_hide_select="0">
```

Comment:
should be the same as above: [Skin],expand_samplers

Reply:
ok, done